### ISSUE TITLE
[BUG] `SystemUpgradeControllerReady` condition manipulation is causing unnecessary reconciliation of the rkecontrolplane object

### ISSUE THREAD

**Issue by Oats87** [rancher/rancher#41202]
**Rancher Server Setup**
- Rancher version: v2.7.2
- Installation option (Docker install/Helm Chart): N/A
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): N/A
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version: N/A
- Cluster Type (Local/Downstream): Downstream provisioned k3s/rke2
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:
N/A

**Describe the bug**
The controller responsible for inspecting the `system-upgrade-controller` Fleet bundle is causing unnecessary thrashing of the condition for `SystemUpgradeControllerReady`.

It is unconditionally (irony intended) setting the condition to `Unknown` and `True`, which is causing the `LastUpdateTime` timestamp to be manipulated unnecessarily. 

For context, if the condition was set to True, when the `rkecontrolplane` object is being processed through the controller, the condition will flap to Unknown then back to True (status based). The call to `Unknown` is unnecessary and causes the `LastUpdateTime` to change.  The code should be changed to only set the `Unknown` status if it is actually unknown.

See: https://github.com/rancher/rancher/blob/dc282bc553b17277c5e7ae1b871857f049842445/pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go#L102

**To Reproduce**
1. Provision a downstream K3s/RKE2 cluster and open a watch outputting as yaml on the `rkecontrolplane` object associated with that cluster. Store the output into a large yaml file, for example,
`kubectl get rkecontrolplane -n fleet-default <cluster-name> -o yaml -w > controlplane.yaml`
1. Restart the `rancher-system-agent` on one of the downstream nodes and wait for the plan application to happen, then wait another minute. 
1. Stop the watch
1. Use the utilities found here: https://gist.github.com/Oats87/c2de1416f1da101c70a4156cf5e48726 to split the yaml i.e. `./split-yaml.sh controlplane.yaml`
1. Diff the yaml using the utilities i.e. `./diff-yaml.sh controlplane.yaml`

**Result**
Observe that the object status changes frequently with the `LastUpdateTime` of the `SystemUpgradeControllerReady` condition being changed often.

For example,
```
Diff of controlplanes.yaml-split/14.yaml and controlplanes.yaml-split/15.yaml:
*** controlplanes.yaml-split/14.yaml	Mon Apr 17 21:13:08 2023
--- controlplanes.yaml-split/15.yaml	Mon Apr 17 21:13:08 2023
***************
*** 24,34 ****
      blockOwnerDeletion: true
      controller: true
      kind: Cluster
      name: testingreg1
      uid: d0c894c2-635e-4211-be82-fcc3a9304609
!   resourceVersion: "526196"
    uid: a9ece278-b76b-4d28-9e3c-663396194d02
  spec:
    chartValues:
      rke2-calico: {}
    clusterName: testingreg1
--- 24,34 ----
      blockOwnerDeletion: true
      controller: true
      kind: Cluster
      name: testingreg1
      uid: d0c894c2-635e-4211-be82-fcc3a9304609
!   resourceVersion: "526198"
    uid: a9ece278-b76b-4d28-9e3c-663396194d02
  spec:
    chartValues:
      rke2-calico: {}
    clusterName: testingreg1
***************
*** 149,159 ****
          preDrainHooks: null
          skipWaitForDeleteTimeoutSeconds: 0
          timeout: 120
    certificateRotationGeneration: 0
    conditions:
!   - lastUpdateTime: "2023-04-18T04:08:38Z"
      reason: Waiting for System Upgrade Controller Bundle Update
      status: "True"
      type: SystemUpgradeControllerReady
    - lastUpdateTime: "2023-04-18T04:08:43Z"
      status: "True"
--- 149,159 ----
          preDrainHooks: null
          skipWaitForDeleteTimeoutSeconds: 0
          timeout: 120
    certificateRotationGeneration: 0
    conditions:
!   - lastUpdateTime: "2023-04-18T04:08:43Z"
      reason: Waiting for System Upgrade Controller Bundle Update
      status: "True"
      type: SystemUpgradeControllerReady
    - lastUpdateTime: "2023-04-18T04:08:43Z"
      status: "True"
 ```

and 
```
Diff of controlplanes.yaml-split/16.yaml and controlplanes.yaml-split/17.yaml:
*** controlplanes.yaml-split/16.yaml	Mon Apr 17 21:13:08 2023
--- controlplanes.yaml-split/17.yaml	Mon Apr 17 21:13:08 2023
***************
*** 24,34 ****
      blockOwnerDeletion: true
      controller: true
      kind: Cluster
      name: testingreg1
      uid: d0c894c2-635e-4211-be82-fcc3a9304609
!   resourceVersion: "526207"
    uid: a9ece278-b76b-4d28-9e3c-663396194d02
  spec:
    chartValues:
      rke2-calico: {}
    clusterName: testingreg1
--- 24,34 ----
      blockOwnerDeletion: true
      controller: true
      kind: Cluster
      name: testingreg1
      uid: d0c894c2-635e-4211-be82-fcc3a9304609
!   resourceVersion: "526209"
    uid: a9ece278-b76b-4d28-9e3c-663396194d02
  spec:
    chartValues:
      rke2-calico: {}
    clusterName: testingreg1
***************
*** 149,159 ****
          preDrainHooks: null
          skipWaitForDeleteTimeoutSeconds: 0
          timeout: 120
    certificateRotationGeneration: 0
    conditions:
!   - lastUpdateTime: "2023-04-18T04:08:43Z"
      reason: Waiting for System Upgrade Controller Bundle Update
      status: "True"
      type: SystemUpgradeControllerReady
    - lastUpdateTime: "2023-04-18T04:08:44Z"
      status: "True"
--- 149,159 ----
          preDrainHooks: null
          skipWaitForDeleteTimeoutSeconds: 0
          timeout: 120
    certificateRotationGeneration: 0
    conditions:
!   - lastUpdateTime: "2023-04-18T04:08:44Z"
      reason: Waiting for System Upgrade Controller Bundle Update
      status: "True"
      type: SystemUpgradeControllerReady
    - lastUpdateTime: "2023-04-18T04:08:44Z"
      status: "True"
***************
*** 164,169 ****
    - lastUpdateTime: "2023-04-17T23:55:42Z"
      status: "True"
      type: Provisioned
    initialized: true
    observedGeneration: 1
-   ready: true
--- 164,168 ----
```

**Expected Result**

**Screenshots**

**Additional context**

It appears this was introduced here: https://github.com/rancher/rancher/pull/40500

SURE-6266

**Comment by snasovich** [comment]
Root cause seems to be https://github.com/rancher/rancher/pull/40500 so assigning to @HarrisonWAffel to address.

**Comment by Oats87** [comment]
adding myself as the assignee as my PR addresses this issue

**Comment by sowmyav27** [comment]
From an offline comment of @Oats87 

> I can give a symptom validation for it (i.e. for example, turn rancher to debug mode and watch the secrets and see thrashing of the condition), and then upgrade and observe that it is not occurring anymore.

And also @Oats87 to file a separate GH issue to track unit tests for this specific issue.

**Comment by Oats87** [comment]
Issue has been filed to track adding a unit test here: https://github.com/rancher/rancher/issues/41495

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|As a standard user, provision a downstream RKE2 custom cluster with highest K8s version and validate that the rkecontrolplane object is not unnecessarily reconciliating|
|2|As a standard user, provision a downstream K3S custom cluster with highest K8s version and validate that the rkecontrolplane object is not unnecessarily reconciliating|
|3|As a standard user, provision a downstream RKE2 custom cluster with second highest K8s version, validate that the rkecontrolplane object is not unnecessarily reconciliating, upgrade cluster K8s version and re-validate the rkecontrolplane object is not unnecessarily reconcillating|
|4|As a standard user, provision a downstream K3S custom cluster with second highest K8s version, validate that the rkecontrolplane object is not unnecessarily reconciliating, upgrade cluster K8s version and re-validate the rkecontrolplane object is not unnecessarily reconcillating|
|5|Setup Rancher v2.7.2, as a standard user, provision a downstream RKE2 custom cluster as a standard user, note that rkecontrolplane object is reconciliating, upgrade Rancher to v2.7-head and validate that the rkecontrolplane objects is not reconciliating|

**Comment by markusewalker** [comment]
Reproduced this issue on `v2.7.2`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/b04c942829d1a0b3711a04e7ae9db027f8bc59c7
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream RKE2 custom cluster with highest K8s version and validate that the rkecontrolplane object is not unnecessarily reconciliating|:x:|

**REPRO STEPS**

**Scenario 1**
1. As a standard user, provisioned a downstream custom RKE2 `v1.25.9+rke2r1` cluster (1 etcd, 1 cp, 1 worker).
2. In a client machine with the `local` cluster's kubeconfig file, ran the following command: `kubectl get rkecontrolplane -n fleet-default <cluster> -o yaml -w > controlplane.yaml`.
     - At the same time, in a separate tab, SSH'ed into the RKE2 cluster's control plane node and ran command: `sudo systemctl rancher-system-agent`.
3. Waited around 1 minute and then terminated the watch command.
4. Ran the following commands:
```
./split-yaml.sh controlplane.yaml
./diff-yaml.sh controlplane.yaml
```
4. Reproduced the issue as discussed in the issue description.

Now that I have reproduced the issue, I will attempt to validate the issue in the latest `v2.7-head`.

**Comment by markusewalker** [comment]
Validated that this is addressed in `v2.7-head` https://github.com/rancher/rancher/commit/b04c942829d1a0b3711a04e7ae9db027f8bc59c7. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/b04c942829d1a0b3711a04e7ae9db027f8bc59c7
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream RKE2 custom cluster with highest K8s version and validate that the rkecontrolplane object is not unnecessarily reconciliating|:white_check_mark:|
|2|As a standard user, provision a downstream K3S custom cluster with highest K8s version and validate that the rkecontrolplane object is not unnecessarily reconciliating|:white_check_mark:|
|3|As a standard user, provision a downstream RKE2 custom cluster with second highest K8s version, validate that the rkecontrolplane object is not unnecessarily reconciliating, upgrade cluster K8s version and re-validate the rkecontrolplane object is not unnecessarily reconcillating|:white_check_mark:|
|4|As a standard user, provision a downstream K3S custom cluster with second highest K8s version, validate that the rkecontrolplane object is not unnecessarily reconciliating, upgrade cluster K8s version and re-validate the rkecontrolplane object is not unnecessarily reconcillating|:white_check_mark:|
|5|Setup Rancher v2.7.2, as a standard user, provision a downstream RKE2 custom cluster as a standard user, note that rkecontrolplane object is reconciliating, upgrade Rancher to v2.7-head and validate that the rkecontrolplane objects is not reconciliating|:white_check_mark:|
|6|Setup Rancher v2.7.2, as a standard user, provision a downstream K3S custom cluster as a standard user, note that rkecontrolplane object is reconciliating, upgrade Rancher to v2.7-head and validate that the rkecontrolplane objects is not reconciliating|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As a standard user, provisioned a downstream custom RKE2 `v1.26.4+rke2r1` cluster (1 etcd, 1 cp, 1 worker).
2. In a client machine with the `local` cluster's kubeconfig file, ran the following command: `kubectl get rkecontrolplane -n fleet-default <cluster> -o yaml -w > controlplane.yaml`.
     - At the same time, in a separate tab, SSH'ed into the RKE2 cluster's control plane node and ran command: `sudo systemctl rancher-system-agent`.
3. Waited around 1 minute and then terminated the watch command.
4. Ran the following commands:
```
./split-yaml.sh controlplane.yaml
./diff-yaml.sh controlplane.yaml
```
5. Validated that the rkecontrolplane is not unnecessarily reconciliating as the output of step 3 does not show any difference between the compared YAML files.
_______
**Scenario 2**
1. Repeated scenario 1, but provisioned a downstream custom K3S `v1.26.4+k3s` instead; all of the successful results were seen for this scenario as well.
_______
**Scenario 3**
1. Repeated scenario 1 steps 1-5, but provisioned a downstream custom RKE2 `v1.25.9+rke2r1` cluster instead.
2. Upgraded the K8s cluster to `v1.26.4+rke2r1`; validated that the cluster and nodes came up as `Active` and that the K8s cluster is updated.
3. Repeated scenario 1 steps 2-4 and validated that no issues were seen.
_______
**Scenario 4**
1. Repeated scenario 1 steps 1-5, but provisioned a downstream custom K3S `v1.25.9+k3s1` cluster instead.
2. Upgraded the K8s cluster to `v1.26.4+k3s1`; validated that the cluster and nodes came up as `Active` and that the K8s cluster is updated.
3. Repeated scenario 1 steps 2-4 and validated that no issues were seen.
_________
**Scenario 5**
1. Repeated scenario 1 steps 1-4, but provisioned a downstream custom RKE2 `v1.25.9+rke2r1` on Rancher v2.7.2.
2. Reproduced the issue as noted in the issue description.
3. Upgraded Rancher `v2.7.2` to `v2.7-head`; validated that the cluster did not auto-upgrade itself.
4. Repeated scenario 1 steps 2-4 and validated that the reported issue no longer appears.
________
**Scenario 6**
1. Repeated scenario 1 steps 1-4, but provisioned a downstream custom K3S`v1.25.9+k3s1` on Rancher v2.7.2.
2. Reproduced the issue as noted in the issue description.
3. Upgraded Rancher `v2.7.2` to `v2.7-head`; validated that the cluster did not auto-upgrade itself.
4. Repeated scenario 1 steps 2-4 and validated that the reported issue no longer appears.
